File: backend/app/schemas/room.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, List
from app.schemas.user import UserResponse
from app.schemas.preference import PreferenceResponse

class RoomBase(BaseModel):
    status: str = "active"

class RoomCreate(BaseModel):
    pass

class GroupMemberResponse(BaseModel):
    room_id: int
    user_id: int
    pref_id: Optional[int] = None
    user: Optional[UserResponse] = None
    preference: Optional[PreferenceResponse] = None

    class Config:
        from_attributes = True

class RoomResponse(RoomBase):
    room_id: int
    host_id: int
    room_code: str
    created_at: datetime
    host: Optional[UserResponse] = None
    members: List[GroupMemberResponse] = []

    class Config:
        from_attributes = True

class RoomJoin(BaseModel):
    room_code: str

File: frontend/src/services/room.ts
import api from './api';
import type { UserResponse } from './auth';
import type { PreferenceResponse } from './preferences';

export interface GroupMemberResponse {
  room_id: number;
  user_id: number;
  pref_id?: number;
  user?: UserResponse;
  preference?: PreferenceResponse;
}

export interface RoomResponse {
  room_id: number;
  host_id: number;
  room_code: string;
  status: string;
  created_at: string;
  host?: UserResponse;
  members: GroupMemberResponse[];
}

export const roomApi = {
  createRoom: async (): Promise<RoomResponse> => {
    const response = await api.post<RoomResponse>('/rooms');
    return response.data;
  },

  joinRoom: async (roomCode: string): Promise<RoomResponse> => {
    const response = await api.post<RoomResponse>('/rooms/join', { room_code: roomCode });
    return response.data;
  },

  getActiveRooms: async (): Promise<RoomResponse[]> => {
    const response = await api.get<RoomResponse[]>('/rooms/active');
    return response.data;
  },

  getRoomDetails: async (roomId: number): Promise<RoomResponse> => {
    const response = await api.get<RoomResponse>(`/rooms/${roomId}`);
    return response.data;
  },

  selectRoomPreference: async (roomId: number, prefId: number): Promise<GroupMemberResponse> => {
    const response = await api.put<GroupMemberResponse>(`/rooms/${roomId}/preference/${prefId}`);
    return response.data;
  },

  dissolveRoom: async (roomId: number): Promise<void> => {
    await api.delete(`/rooms/${roomId}`);
  },
};
